crates/subspace-farmer/src/ws_rpc_server.rs

```diff
@@ -1,16 +1,124 @@
+use crate::object_mappings::ObjectMappings;
 use crate::plot::Plot;
 use async_trait::async_trait;
+use hex_buffer_serde::{Hex, HexForm};
 use jsonrpsee::proc_macros::rpc;
 use jsonrpsee::types::error::Error;
-use log::debug;
-use subspace_core_primitives::Piece;
+use log::{debug, error};
+use parity_scale_codec::{Compact, CompactLen, Decode, Encode};
+use serde::{Deserialize, Serialize};
+use std::ops::{Deref, DerefMut};
+use subspace_archiving::archiver::{Segment, SegmentItem};
+use subspace_core_primitives::{Piece, Sha256Hash, PIECE_SIZE};
 use subspace_solving::SubspaceCodec;
 
+/// Maximum expected size of one object in bytes
+const MAX_OBJECT_SIZE: usize = 5 * 1024 * 1024;
```

Comment:
Where does this limit come from? I thought It should be limited by the transaction size?

Reply:
This is just a protection against reading infinite number of segments into the future if there is a logic error somewhere. Farmer library doesn't write data to the blockchain.